cmd/consumer/main.go

```diff
@@ -46,23 +46,14 @@ func consumeEvent(event cloudevents.Event) error {
 		return fmt.Errorf("error unmarshalling json: %w", err)
 	}
 
-	// deserialize the request
-	r := bufio.NewReader(strings.NewReader(data.Req))
-	req, err := http.ReadRequest(r)
-	if err != nil {
-		return fmt.Errorf("problem reading request: %w", err)
-	}
 	// client for sending request
 	client := &http.Client{}
-
-	// build new url - writing the request removes the URL and places in URI.
-	req.URL, err = url.Parse("http://" + req.Host + req.RequestURI)
-	if err != nil {
-		return fmt.Errorf("Problem parsing url: %w", err)
+	req, err := http.NewRequest(data.ReqMethod, data.ReqURL, nil)
```

Comment:
What happens if there is an error when creating a `NewRequest`?

Reply:
just updated with an error check, thanks.